compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/TypeParamResolver.java

```diff
@@ -267,4 +324,67 @@ public BType visit(BTypeReferenceType typeInSymbol, BType s) {
     private boolean isTypeParam(BType type) {
         return type == this.typeParam;
     }
+
+    private BVarSymbol createNewVarSymbol(BVarSymbol symbol, BType newType) {
+
+        if (symbol == null) {
+            return null;
+        }
+
+        BType originalType = symbol.getType();
+
+        if (originalType == newType) {
+            return symbol;
+        }
+
+        BVarSymbol duplicateSymbol = duplicateSymbol(symbol);
+        duplicateSymbol.type = newType;
+
+        return duplicateSymbol;
+    }
+
+    private BInvokableSymbol resolveInvokableSymbol(BInvokableSymbol symbol, BType newInvokableType, BType boundType) {
+        BInvokableSymbol newInvokableSymbol = duplicateSymbol(symbol);
+        newInvokableSymbol.type = newInvokableType;
+
+        for (BVarSymbol param : symbol.params) {
+            BType newParamType = resolve(param.type, boundType);
+            BVarSymbol newVarSymbol = createNewVarSymbol(param, newParamType);
+            newInvokableSymbol.params.add(newVarSymbol);
+        }
+        if (symbol.restParam != null) {
```

Comment:
```suggestion
        }
        
        if (symbol.restParam != null) {
```
Shouldn't there be a new line here?